python/lsst/ts/observatory/control/auxtel/latiss.py

```diff
@@ -0,0 +1,551 @@
+            linear_stage=linear_stage,
+            group_id=group_id,
+            checkpoint=checkpoint,
+        )
+
+    async def take_engtest(
+        self,
+        exptime,
+        n=1,
+        filter=None,
+        grating=None,
+        linear_stage=None,
+        group_id=None,
+        checkpoint=None,
+    ):
+        """Take a series of engineering test images.
+
+        Parameters
+        ----------
+        exptime : `float`
+            Exposure time for flats.
+        n : `int`
+            Number of frames to take.
+        filter : `None` or `int` or `str`
+            Filter id or name. If None, do not change the filter.
+        grating : `None` or `int` or `str`
+            Grating id or name.  If None, do not change the grating.
+        linear_stage : `None` or `float`
+            Linear stage position.  If None, do not change the linear stage.
+        group_id : `str`
+            Optional group id for the data sequence. Will generate a common
+            one for all the data if none is given.
+        checkpoint : `coro`
+            A optional awaitable callback that accepts one string argument
+            that is called before each bias is taken.
+
+        """
+        return await self.take_imgtype(
+            imgtype="ENGTEST",
+            exptime=exptime,
+            n=n,
+            filter=filter,
+            grating=grating,
+            linear_stage=linear_stage,
+            group_id=group_id,
+            checkpoint=checkpoint,
+        )
+
+    async def take_imgtype(
+        self,
+        imgtype,
+        exptime,
+        n,
+        filter=None,
+        grating=None,
+        linear_stage=None,
+        group_id=None,
+        checkpoint=None,
+    ):
+        """Take a series of images of the specified image type.
+
+        Parameters
+        ----------
+        exptime : `float`
+            Exposure time for flats.
+        n : `int`
+            Number of frames to take.
+        filter : `None` or `int` or `str`
+            Filter id or name. If None, do not change the filter.
+        grating : `None` or `int` or `str`
+            Grating id or name.  If None, do not change the grating.
+        linear_stage : `None` or `float`
+            Linear stage position.  If None, do not change the linear stage.
+        checkpoint : `coro`
+            A optional awaitable callback that accepts one string argument
+            that is called before each bias is taken.
+
+        """
+
+        if imgtype not in self.valid_imagetype:
+            raise RuntimeError(
+                f"Invalid imgtype:{imgtype}. Must be one of "
+                f"{self.valid_imagetype!r}"
+            )
+
+        exp_ids = np.zeros(n, dtype=int)
+
+        if group_id is None:
+            self.log.debug("Generating group_id")
+            group_id = self.next_group_id()
+
+        if imgtype == "BIAS" and exptime > 0.0:
+            self.log.warning("Image type is BIAS, ignoring exptime.")
```

Comment:
I would gently suggest moving these tests to `expose` because it is a public API. I realize that it's a bit slower to test for each image, but all public APIs should fully test their arguments.

At the same time I agree it's nice to catch errors early. But I _think_ it's "early enough" in `expose` and it probably beats duplicating the test code.

Also I gently suggest changing exptime to 0 here (in this if statement, whether it gets moved to `expose` or not). That saves you having to test it again later when you use it (and possibly forgetting to do so).